packages/block-library/src/query-no-results/index.php

```diff
@@ -21,22 +21,23 @@ function render_block_core_query_no_results( $attributes, $content, $block ) {
 
 	$page_key   = isset( $block->context['queryId'] ) ? 'query-' . $block->context['queryId'] . '-page' : 'query-page';
 	$page       = empty( $_GET[ $page_key ] ) ? 1 : (int) $_GET[ $page_key ];
-	$query_args = build_query_vars_from_query_block( $block, $page );
 	// Override the custom query with the global query if needed.
 	$use_global_query = ( isset( $block->context['query']['inherit'] ) && $block->context['query']['inherit'] );
 	if ( $use_global_query ) {
 		global $wp_query;
-		if ( $wp_query && isset( $wp_query->query_vars ) && is_array( $wp_query->query_vars ) ) {
-			$query_args = wp_parse_args( $wp_query->query_vars, $query_args );
-		}
+		$query = $wp_query;
+	} else {
+		$query_args = build_query_vars_from_query_block( $block, $page );
+		$query      = new WP_Query( $query_args );
 	}
-	$query = new WP_Query( $query_args );
 
-	if ( $query->have_posts() ) {
+	if ( ! $query->have_posts() ) {
```

Comment:
We shouldn't change this logic. This block is actually a block of Query Loop and a way to show specific content/blocks if the Query doesn't have any results. So the previous check is correct to render nothing if the used query has results. Makes sense?

Reply:
Ah this is a mistake. So we render this block even if there are results. Got it.

Reply:
```suggestion
	if ( $query->have_posts() ) {
```